### ISSUE TITLE
Connect's logic for app cert issue causes`tsh status` to report incorrect "Valid until" time when TTL is low

### ISSUE THREAD

**Issue by ravicious** [gravitational/teleport#42291]
1. Log in as a user with a role that has TTL set to `1m`.
2. Run `tsh status` and note the time reported until "Valid until".
3. Set up a connection to an app to issue a new app cert.
4. Run `tsh status` again, it should report a different expiration time.

It just seems to pick the expiration time from a wrong cert, as the user cert actually expires after the original time reported by the first run of `tsh status`. Likely caused by perhaps a superfluous call to `ReissueUserCerts` (needs investigation, see https://github.com/gravitational/teleport/issues/42291#issuecomment-2160195911).

https://github.com/user-attachments/assets/62b91dfe-178d-4355-b668-925fa0e1a145

**Comment by Joerger** [comment]
I wasn't able to reproduce this myself, so there may have been an additional factor missing from the repro.

**Comment by ravicious** [comment]
You're right, I was testing this through tsh and Connect and I must've not noticed that `tsh proxy app dumper` actually doesn't cause this behavior at all. It only exists in Connect, both in VNet and local proxies which share the same cert creation logic.

I believe the bug lies here:

https://github.com/gravitational/teleport/blob/508109b81a40e27c7412d6b73adf562394329182/lib/teleterm/clusters/cluster_apps.go#L151-L163

Before we do the actual `clusterClient.IssueUserCertsWithMFA` call, we reissue user certs. I don't know exactly why we do it. I remember we used to have problems with db access without this call and then we copied it over to other resources as well.

I'm going to mark it as not a test plan problem then, thanks.

**Comment by ravicious** [comment]
This might have been fixed by #44043.

**Comment by zmb3** [comment]
Could this have been fixed by #49732?

**Comment by ravicious** [comment]
Most likely not, as that PR is about Web UI sessions rather than sessions for apps.
